coordinator/persistence-dse-next/src/main/java/io/stargate/db/cassandra/impl/RowDecoratorImpl.java

```diff
@@ -0,0 +1,136 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.db.cassandra.impl;
+
+import io.stargate.db.AbstractRowDecorator;
+import io.stargate.db.ComparableKey;
+import io.stargate.db.datastore.Row;
+import io.stargate.db.schema.TableName;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import org.apache.cassandra.db.Clustering;
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.db.PartitionPosition;
+import org.apache.cassandra.dht.ByteOrderedPartitioner;
+import org.apache.cassandra.dht.Murmur3Partitioner;
+import org.apache.cassandra.dht.OrderPreservingPartitioner;
+import org.apache.cassandra.dht.Token;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+import org.apache.cassandra.utils.bytecomparable.ByteSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RowDecoratorImpl extends AbstractRowDecorator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RowDecoratorImpl.class);
+
+  private final TableMetadata metadata;
+
+  public RowDecoratorImpl(TableName tableName) {
+    this(tableName, Conversion.toTableMetadata(tableName));
+  }
+
+  private RowDecoratorImpl(TableName tableName, TableMetadata tableMetadata) {
+    super(
+        tableName,
+        tableMetadata.partitionKeyColumns().stream()
+            .map(c -> c.name.toCQLString())
+            .collect(Collectors.toList()));
+    this.metadata = tableMetadata;
+  }
+
+  private DecoratedKey decoratedKeyFromRawKeyValues(Object... rawKeyValues) {
+    Clustering<?> key = metadata.partitionKeyAsClusteringComparator().make(rawKeyValues);
+    return metadata.partitioner.decorateKey(key.serializeAsPartitionKey());
+  }
+
+  @Override
+  protected ComparableKey<?> decoratePrimaryKey(Object... rawKeyValues) {
+    DecoratedKey decoratedKey = decoratedKeyFromRawKeyValues(rawKeyValues);
+    return new ComparableKey<>(PartitionPosition.class, decoratedKey);
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p><b>IMPORTANT:</b> we only support getting the bytes for the Murmur3Partitioner,
+   * ByteOrderedPartitioner & OrderPreservingPartitioner. Any other partitioner will only depend on
+   * the key comparison, which is not going to work for the correct OR queries in the Docs API V2.
+   */
+  @Override
+  public ByteBuffer getComparableBytes(Row row) {
+    DecoratedKey decoratedKey = decoratedKeyFromRawKeyValues(primaryKeyValues(row));
+    ByteComparable.Version version = ByteComparable.Version.OSS41;
```

Comment:
Any reason why this needed to be changed to `OSS41`?

Reply:
@tatu would know

Reply:
I think this was copied from impl by Vector Search team (and/or Jeremiah) -- `persistence-dse-db-all` of cndb --  I do not know what the semantics are. But I do know that while it was forked from 4.0.7, it contains most (... but not all) 4.1.x changes.

Reply:
Looking at `dse-db-all`, `ByteComparable.Version` only has following values:

```
    public static enum Version {
        LEGACY,
        OSS41;
    }
```

for what that is worth.

But I am not 100% sure how these should be merged: it does look like most of this class is from persistence-cassandra-4.0; and not from CNDB/vector team's impl.

Reply:
Filed https://github.com/stargate/stargate/issues/2693 as a follow-up. It may be that there is no functional differences, just slightly different formatting etc but I'll verify.